Content.Shared/Audio/AudioHelpers.cs

```diff
@@ -16,6 +18,41 @@ public static AudioParams WithVariation(float amplitude)
             return AudioParams.Default.WithPitchScale(scale);
         }
 
+        // Might as well just hardcode these because the audio system is limited to pitching up and down
+        // by 12 semitones anyway (ie. 0.5 to 2.0 multiplier).
+        private static readonly float[] SemitoneMultipliers =
+        {
+            0.5f, 233.08f/440f, 246.94f/440f, 261.63f/440f,
+            277.18f/440f, 293.66f/440f, 311.13f/440f, 329.63f/440f,
+            349.23f/440f, 369.99f/440f, 392.00f/440f, 415.30f/440f,
+            1.0f,
+            466.16f/440f, 493.88f/440f, 523.25f/440f, 554.37f/440f,
+            587.33f/440f, 622.25f/440f, 659.26f/440f, 698.46f/440f,
+            739.99f/440f, 783.99f/440f, 830.61f/440f, 2.0f
+        };
```

Comment:
I'm not sure how this would go with actual music. MIDIs and effects might be fine.
The problem with this is pianos for example have equal temperament tuning so a note an octave apart isn't exactly double / half.

Reply:
I just went off of the single bit of music theory trivia I remember from high school, which is that going up an octave doubles the pitch, and this table on Wikipedia: https://en.wikipedia.org/wiki/Pitch_(music)#Labeling_pitches

Reply:
Yeah if they're pure tones it'll work, even if they're tuned to 444HZ I think. According to wikipedia there might be equal tempered midi? 

https://en.wikipedia.org/wiki/MIDI_tuning_standard

> The first byte of the frequency data word specifies the highest equal-tempered semitone not exceeding the frequency.

If midi can have equal temperament then it means any coordinated midi music for a band will sound out of tune if it's varied.

Reply:
Shouldn't really matter since this is mainly supposed to be used for sound effects. MIDIs have their own subsystem.

Reply:
> equal temperament tuning so a note an octave apart isn't exactly double / half

equal temperament is dividing the octave into 'equal' parts (as in divide/multiply by Xth root of 2 to get the previous/next semitone), the factor of an octave is still 2.

Reply:
Okay octave was a poor example but for stuff like perfect 5ths etc it still applies.

Reply:
I don't see a reason to use these for sound effects when the pitch helper already exists but if you foresee using it for stuff other than the horn I'll merge it.

Reply:
The original inspiration is Oxygen Not Included doing this with the sound effect they play when placing building ghosts. It adds unique flair to repeated sound effects because it makes them sound more musical than random out-of-tune pitch variation, especially if the sound effect is in tune to begin with. So that's the kind of thing it could be used for.

Reply:
Merged. 2 random thoughts for future
1. Pentatonic scale is more suited for this IMO.
2. Chuck it on doors too.